api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -751,21 +753,34 @@ public User getUserByActivationKey(String activationKey) {
 	}
 	
 	/**
+	 * @throws MessageException
 	 * @see org.openmrs.api.UserService#setUserActivationKey(org.openmrs.User)
-	 * 
 	 */
 	@Override
 	public User setUserActivationKey(User user) {
 		String token = RandomStringUtils.randomAlphanumeric(20);
 		long time = System.currentTimeMillis() + getValidTime();
 		String hashedKey = Security.encodeString(token);
-		String activationKey = hashedKey+":"+time;
+		String activationKey = hashedKey + ":" + time;
 		LoginCredential credentials = dao.getLoginCredential(user);
 		credentials.setActivationKey(activationKey);	
 		dao.setUserActivationKey(credentials);	
-			
-		//Send Email with unhashed  activation key and Date:
 		
+		String link = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GP_HOST_URL)
+		        .replace("activationkey", "activationkey=" + token);
+		String msg = "Hi " + user.getUsername()
+		        + ",\n It looks like you want to change your password, click on the following link to do so.\n"
+		        + link
+		        + "\n Please disregard this email if you didn't innitiate the password reset process.  \n The openmrs Team";
+		
+		try {
+			Message mail = Context.getMessageService().createMessage(user.getEmail(),
+			    Context.getAdministrationService().getGlobalProperty("mail.from"), "Password Reset Request", msg);
+			Context.getMessageService().sendMessage(mail);
+		}
+		catch (MessageException e) {
+			// Prevent propagation of exception to the webservice should in case message was not sent  
```

Comment:
You shouldn't be swallowing the exception

Reply:
In the restwebservice A post for request of passwordreset  will trigger this method and that post is supposed to always return http.OK  and if the exception is not handled here it will propergate back as the response and we end up not having an ok response.

Reply:
You should not swallow the exception here, that is implementation logic in REST and should have nothing to do with the service layer